### DISCUSSION TITLE
Support for localization of messages

### DISCUSSION THREAD

**Discussion by toniwiki** [fastapi/typer#753] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
python -m my-cli-app --help
```

### Description

I am interested in creating a CLI application with Typer that supports multiple languages. To do this I want to be able to localize automatically generated default strings in help, error messages, etc.

I am new to Typer (and Python in general) but I think I understand that Click is the one that wraps the internal strings with gettext. However, it is not clear to me how I can change the default messages from Typer.

Could someone give me a simple example of how to do it?

Thanks

### Operating System

Linux

### Typer Version

0.7.0

### Python Version

3.10.9

**Comment by svlandeg** [comment, marked as answer]
Hi!

Both Click and Typer provide support for localization of messages, by wrapping hard-coded strings in the `gettext` wrapper as you can see in https://github.com/tiangolo/typer/blob/master/typer/core.py, e.g.
```
from gettext import gettext as _

if self.required:
    extra.append(_("required"))
```

(cf also relevant PRs for [Click](https://github.com/pallets/click/pull/1829) and [Typer](https://github.com/tiangolo/typer/pull/317))

What this means is that eventually, you would be able to provide a translation of `"(required)"` in your own language, store that in a compiled dictionary, change the locale to your language, and then eventually Typer will print the translated version of this string in its help description for instance.

To use this feature, you'll need to set up a `locales` directory, add subdirectories for the languages you want to support, identify all translatable strings in your function, and finally create the translations. There's some tutorials about this that go beyond the scope of this discussion forum so I'll link them here, you can probably find others as well: [[1]](https://phrase.com/blog/posts/translate-python-gnu-gettext/), [[2]](https://crowdin.com/blog/2022/09/28/python-app-translation-tutorial).

**Reply by toniwiki** [reply]
Thank you very much for your help @svlandeg.

Now I can use internationalization for strings in my project with no problem, but I don't know how to instruct my application to perform translation of strings generated from dependencies like Typer.

I have directly included the translation of Typer strings in my localization files but these are not applied in the messages generated from the Typer core.

Any help on where I should look?
